00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/hub.yaml proposal for themes.md

```diff
@@ -0,0 +1,303 @@
+	    miscellaneous:
+		    - A minimalistic theme perfect for focusing on writing
+		    - Opinionated but highly customizable
+
+```
+
+These two were not included in the `features` list:
+
+- ~~has style settings~~ -> we can detect this automatically with our script, so no need to mention this manually
+- ~~plugin_support - plugin compatibility list~~ -> Moved to its own section
+
+Listing features in the `hub.yaml` should be extremely flexible. For the hub to recognize your feature, you can use:
+
+- A boolean
+  ```yaml
+  theme:
+    features: 
+          callouts: true
+   ```
+- A path to one or many screenshots on your repo
+  ```yaml
+  theme:
+    features: 
+          callouts:
+            screenshots: # Path to screenshots for this feature (from the root of the repo)
+                - file: docs/all_dnd_callouts.png
+                  caption: All the D&D callouts for My Theme
+                - file: docs/treasure_dnd_callout.png
+                  caption: Example of the treasure D&D callout
+   ```
+- A path to a markdown document to be included as is in the hub
+  ```yaml
+  theme:
+    features: 
+          custom_callouts:
+            docs:
+               file: docs/dnd_callouts_overview.md # from root of repo, to be included as is
+   ```
+- A link to your own documentation
+  ```yaml
+  theme:
+    features: 
+          custom_callouts:
+            docs:
+		       links: # List of strings (can be markdown-formatted)
+		           - [More information](www.mytheme.com/dnd_callouts)
+   ```
+- Combinations of the above options (except boolean)
+  ```yaml
+  theme:
+    features: 
+          custom_callouts:
+            docs:
+               file: docs/dnd_callouts_overview.md # from root of repo, to be included as is
+		       links: # List of strings (can be markdown-formatted)
+		           - [More information](www.mytheme.com/dnd_callouts)
+            screenshots: # Path to screenshots for this feature (from the root of the repo)
+                - file: docs/all_dnd_callouts.png
+                  caption: All the D&D callouts for My Theme
+                - file: docs/treasure_dnd_callout.png
+                  caption: Example of the treasure D&D callout
+                  
+   ```
+
+%% Note: for the implementation we'd need to know how to structure the templates so that combinations of these make sense %%
+
+### Plugin support
+
+This key (`plugin_support`) is essentially an alternative to parsing the [compatible plugins](https://publish.obsidian.md/hub/04+-+Guides%2C+Workflows%2C+%26+Courses/Guides/Adding+plugin+compatibility+for+themes+to+the+Obsidian+Hub) from the `obsidian.css` file if you prefer to have the hub stuff outside of your theme file. If you don't specify this key, we'll still try to read your plugins from the css file.
+
+```yaml
+theme:
+    plugin_support: [] # IDs of the plugins that have been styled by this theme (adding them here would ignore the plugin IDs on the obsidian.css file)
```

Comment:
Will people know how if strings need to be quoted? I presume they would be comma-separated...